### PR TITLE
feat: introduce textarea variants placeholders

### PR DESCRIPTION
**Pull request by devodii**

### CODE REVIEW

File: libs/ui/design-system/src/lib/textarea.tsx [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+import * as TextareaPrimitive from '@nmit-coursition/ui/primitives/textarea'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
 
-interface TextareaWithTextProps extends React.ComponentProps<'textarea'> {
+const textAreaVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      invalid: '',
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/textarea.tsx:10]: **suggestion:** No invalid or valid, it's based on validity of input/textarea.

File: libs/ui/design-system/src/lib/textarea.tsx [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+      outlined: '',
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/textarea.tsx:11]: **question:** Where is outlined defined? It's nowhere in props in Figma. TextArea has no variants, so it doesn't need CVA at all. :)

image

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/design-system/src/lib/textarea.tsx (hunk 1 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/textarea.tsx:10]: Ookay

Comment (devodii) [reply libs/ui/design-system/src/lib/textarea.tsx:10]: I'm creating an API for validations.

File: libs/ui/primitives/src/textarea.tsx [hunk 3]
```diff
@@ -2,9 +2,9 @@ import * as React from 'react'
 
 import { cn } from '@nmit-coursition/ui/utils'
 
-export interface TextareaProps extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
+export interface RootProps extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
 
-const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(({ className, ...props }, ref) => {
+const Main = React.forwardRef<HTMLTextAreaElement, RootProps>(({ className, ...props }, ref) => {
```
Comment (BleedingDev) [inline libs/ui/primitives/src/textarea.tsx:7]: **suggestion:** We should never touch `primitives` unless Shadcn updates it. It can cause conflicts.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**